ts/builder.ts

```diff
@@ -372,6 +372,7 @@ export class Builder {
       const trimmed_size = i + 1;
   
       // Write out the current vtable.
+      i = this.vtable_in_use - 1;
```

Comment:
done.

Reply:
The comment I've added is: ` // Add the offset of every field.`

Which is exactly the issue this fixed. There are cases, like the one demonstrated where some field is NOT represented in the vtable. Which is a severe issue, right?

I would like to understand more the block that was causing this problem which is the block above. The one that calculates the `trimmed_size`. What is it about? What's the rationale of such `trimmed_size`?

Reply:
@dbaileychess, now I understand the role of `trimmed_size`.

Its purpose is to save space in the vtable when fields are not defined, in such a way that there is no offset written for  undefined fields starting from the last one defined in the schema, until a defined field is found.

Thus, my previous assumption was wrong, and the buffer is correctly created. The problem is within `requiredField()` builder method, which does NOT verify that the offset of the field is smaller than the size of the vtable. The issue is fixed in this PR now.

NOTE: Contrary to what I wrote previously, C++ is not affected and works correctly. I was not getting the assert because the compilation flag NDEBUG was being set on my setup. So the problem only affects TS and not C++.